dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/Validator.java

```diff
@@ -42,11 +48,30 @@
      */
     void validate( ValidationErrorReporter reporter, TrackerBundle bundle, T input );
 
+    /**
+     * Indicates whether this {@link Validator} should be run given the
+     * {@link TrackerImportStrategy}.
+     *
+     * @param strategy import strategy
+     * @return true if this validator should be run and false otherwise
+     */
     default boolean needsToRun( TrackerImportStrategy strategy )
     {
         return strategy != TrackerImportStrategy.DELETE;
     }
 
+    /**
+     * A signal to the orchestration of the validation that subsequent
+     * {@link Validator}s should not be run if this one fails i.e. adds an
+     * error.
+     * <p>
+     * Note: this mechanism is going to be replaced by
+     * {@link org.hisp.dhis.tracker.validation.hooks.Seq}.
+     * </p>
+     *
+     * @return true if subsequent validators should be skipped on error and
+     *         false otherwise
+     */
     default boolean skipOnError()
```

Comment:
Should this be called "breakOnError" or "exitOnError" since we're not actually skipping "this" validation, but rather subsequent ones, like breaking out of a loop?

Reply:
This one will be removed once we are done migrating all Validators 😄